src/goes/goes_latlon.py

```diff
@@ -79,68 +92,126 @@ def _calc_latlon(self):
         elevation_angle = elevation_angle.reshape(len(elevation_angle) * len(elevation_angle))
-        yaw_flip_flag = source_dataset.variables['yaw_flip_flag'][0]
-        if not yaw_flip_flag:
-            scan_angle = scan_angle[::-1]
-            elevation_angle = elevation_angle[::-1]
-        source_dataset.close()
+        scan_angle = self._goes_util.filter_data_array_by_yaw_flip_flag(scan_angle)
+        elevation_angle = self._goes_util.filter_data_array_by_yaw_flip_flag(elevation_angle)
+        scan_angle = self._filter_by_fill_value(scan_angle)
+        elevation_angle = self._filter_by_fill_value(elevation_angle)
         return scan_angle, elevation_angle
 
+    @staticmethod
+    def _calc_scan_position(latitude):
+        """
+        Calculates the scan position...
+        """
+        scan_position = np.full(latitude.shape, 1, dtype='int32')
+        return scan_position
+
+    def _calc_sensor_zenith_azimuth_view_angles(self, latitude, longitude):
+        """
+        Calculates the sensor zenith, azimuth, and view angles. These angles are limited to values between -80 and 80
+        degrees due to a constraint in CRTM.
+        """
+        goes_imager_projection = self._source_dataset.variables['goes_imager_projection']
+        r_eq = goes_imager_projection.getncattr('semi_major_axis')
+        h = goes_imager_projection.getncattr('perspective_point_height') + \
+            goes_imager_projection.getncattr('semi_major_axis')
+        lat_0 = goes_imager_projection.getncattr('latitude_of_projection_origin')
+        lon_0 = goes_imager_projection.getncattr('longitude_of_projection_origin')
+        h_sqr = np.power(h, 2.0)
+        r_eq_sqr = np.power(r_eq, 2.0)
+        beta = np.arccos(np.cos(latitude - lat_0) * np.cos(longitude - lon_0))
+        sqrt_comp = h_sqr + r_eq_sqr - (2.0 * h * r_eq * np.cos(beta))
+        sensor_zenith_angle = np.arcsin((h * np.sin(beta)) / np.sqrt(sqrt_comp)) * 180.0 / np.pi
+        sensor_azimuth_angle = np.arcsin(np.sin(lon_0 - longitude) / np.sin(beta)) * 180.0 / np.pi
+        sensor_view_angle = sensor_zenith_angle
+        sensor_zenith_angle = self._goes_util.filter_data_array_by_yaw_flip_flag(sensor_zenith_angle)
+        sensor_azimuth_angle = self._goes_util.filter_data_array_by_yaw_flip_flag(sensor_azimuth_angle)
+        sensor_view_angle = self._goes_util.filter_data_array_by_yaw_flip_flag(sensor_view_angle)
+        sensor_zenith_angle = np.nan_to_num(sensor_zenith_angle, nan=-999)
+        sensor_azimuth_angle = np.nan_to_num(sensor_azimuth_angle, nan=-999)
+        sensor_view_angle = np.nan_to_num(sensor_view_angle, nan=-999)
+        sensor_zenith_angle = np.where(sensor_zenith_angle > 80, 80, sensor_zenith_angle)
+        sensor_zenith_angle = np.where(sensor_zenith_angle < -80, -80, sensor_zenith_angle)
+        sensor_azimuth_angle = np.where(sensor_azimuth_angle > 80, 80, sensor_azimuth_angle)
```

Comment:
All azimuth angle is not limited to -80 to 80, instead, it goes from 0 to 360.
```
.//src/CRTM_Parameters.f90:  REAL(fp), PUBLIC, PARAMETER :: MAX_SENSOR_AZIMUTH_ANGLE = 360.0_fp
.//src/CRTM_Parameters.f90:  REAL(fp), PUBLIC, PARAMETER :: MAX_SOURCE_AZIMUTH_ANGLE = 360.0_fp
```

Reply:
OK. I'll change that acceptable range.